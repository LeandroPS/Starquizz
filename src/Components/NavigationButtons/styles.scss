@import '../../styles/common.scss';

.navigation-buttons{
    width: fit-content;

    button{
        outline: none;
    }

    button:nth-child(1){
        margin-right: 10px;
        border: none;
        font: inherit;
        color: $black;
        cursor: pointer;

        &:hover{
            text-decoration: underline;
        }
    }

    button:nth-child(2){
        @extend .button;
    }
}