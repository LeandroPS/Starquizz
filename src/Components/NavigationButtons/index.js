import React from 'react';
import PropTypes from 'prop-types';


import './styles.scss';

const NavigationButtons = ({onNextPage, onPreviousPage}) => {

    return(
        <div className="navigation-buttons">
            <button onClick={onPreviousPage}>
                Anterior
            </button>
            <button onClick={onNextPage}>
                Pr√≥ximo
            </button>
        </div>
    );
}

NavigationButtons.propTypes = {
    onNextPage: PropTypes.func.isRequired,
    onPreviousPage: PropTypes.func.isRequired
}

export default NavigationButtons;