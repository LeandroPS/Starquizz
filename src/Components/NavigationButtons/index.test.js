import React from 'react';
import { create } from "react-test-renderer";
import { shallow } from 'enzyme';
import NavigationButtons from './';

describe('Navigation Buttons', () => {

    const props = {
        onNextPage: jest.fn(),
        onPreviousPage: jest.fn()
    }

    it('renders correctly', () => {
        const tree = create(<NavigationButtons {...props}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('executes onNext function when click on next button', () => {
        const component = shallow(<NavigationButtons {...props}/>);
        const button = component.find('button').last();
        button.simulate('click');

        expect(props.onNextPage).toHaveBeenCalled();
    });

    it('executes onPrevious function when click on Previous button', () => {
        const component = shallow(<NavigationButtons {...props}/>);
        const button = component.find('button').first();
        button.simulate('click');

        expect(props.onPreviousPage).toHaveBeenCalled();
    });

});
