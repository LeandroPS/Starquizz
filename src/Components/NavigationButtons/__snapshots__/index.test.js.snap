// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Buttons renders correctly 1`] = `
<div
  className="navigation-buttons"
>
  <button
    onClick={[MockFunction]}
  >
    Anterior
  </button>
  <button
    onClick={[MockFunction]}
  >
    Pr√≥ximo
  </button>
</div>
`;
